app/controllers/repos_controller.rb

```diff
@@ -138,4 +143,81 @@ def find_benchmark_type_by_category(category)
   def fetch_categories
     @repo.benchmark_types.pluck(:category)
   end
+
+  # @param context indicates if we are calling from a 'releases' or 'commits' context
+  def generate_json(context)
+    ruby_versions = construct_ruby_version(context)
+
+    @charts.map do |chart|
+      # rename for clarity
+      result_data = chart[0]
+      result_type = chart[1]
+
+      # parse the data (sometimes there's two sets of data for one chart)
+      datapoints = JSON.parse(result_data[:columns]).map { |column| column['data'] }
+
+      # this is for when there are two data sets in one chart (ex. rails commits benchmarks)
+      # then the variations will be `with_prepared_statements` and `without_prepared_statements`
+      variations = JSON.parse(result_data[:columns]).map { |column| column['name'] }
+
+      # generate the json
+      config = {
+        benchmark_name: params[:result_type],
+        datapoints: datapoints,
+        ruby_versions: ruby_versions[0],
+        measurement: result_type[:name],
+        unit: result_type[:unit]
+      }
+      config[:variations] = variations if variations.length > 1
+      config
+    end
+  end
+
+  # @param context indicates if we are calling from a 'releases' or 'commits' context
+  def construct_ruby_version(context)
```

Comment:
Can you move `construct_ruby_version` and `generate_json` methods into separate class and cover those with tests?